### PR TITLE
Add rpc api add on server

### PR DESCRIPTION
**Pull request by vderunov**
Screenshot 2024-03-21 at 18 09 50
Screenshot 2024-03-21 at 18 08 18

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -177,6 +185,25 @@ app.get('/protected', (req, res) => {
...
+
+  createProxyMiddleware({
+    target: IPFS_URL,
+    onProxyReq: (proxyReq) => {
+      proxyReq.setHeader('Content-Type', formData.getHeaders()['content-type']);
+      formData.pipe(proxyReq);
+    },
+  })(req, res, next);
+});
+
```
Comment (noisekit) [inline server.js:206]: Though it is good for this PR, for the next one we should avoid local uploads and stream to the backend directly as is.

Something like this.
```suggestion
app.use('/api/v0/add', createProxyMiddleware({
    target: IPFS_URL,
    selfHandleResponse: true
}));
```

Same for `/cat`.

**Review by noisekit — APPROVED** [review]

**Review by noisekit — APPROVED** [review]
In the next (separate) PR please add authorisation for `/api/v0/add`

File: server.js [hunk 2]
```diff
@@ -177,6 +165,8 @@ app.get('/protected', (req, res) => {
   res.send('Hello! You are viewing protected content.');
 });
 
+app.use('/api/v0/add', createProxyMiddleware({ target: IPFS_URL }));
```
Comment (noisekit) [inline server.js:168]:
```suggestion
app.use('/api/v0/add', authenticateTokenб createProxyMiddleware({ target: IPFS_URL }));
```
maybe this?

Comment (vderunov) [reply server.js]: yes, fixed

**Pull request merged**